compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -5838,10 +5740,72 @@ public void visit(BLangGroupExpr groupExpr) {
     @Override
     public void visit(BLangRecordLiteral recordLiteral) {
         List<RecordLiteralNode.RecordField> fields =  recordLiteral.fields;
+        updateFieldsOfRecordLiteral(recordLiteral, fields);
         fields.sort((v1, v2) -> Boolean.compare(isComputedKey(v1), isComputedKey(v2)));
         result = rewriteExpr(rewriteMappingConstructor(recordLiteral));
     }
 
+    public void visit(BFunctionPointerInvocation functionPointerInvocation) {
+        result = functionPointerInvocation;
+    }
+
+    private List<String> getNamesOfRecordFields(List<RecordLiteralNode.RecordField> fields) {
+        List<String> fieldNames = new ArrayList<>();
+        for (RecordLiteralNode.RecordField field : fields) {
+            if (field.getKind() == NodeKind.RECORD_LITERAL_KEY_VALUE) {
+                BLangExpression key = ((BLangRecordLiteral.BLangRecordKeyValueField) field).key.expr;
+                if (key.getKind() == NodeKind.LITERAL) {
+                    fieldNames.add(((BLangLiteral) key).originalValue);
+                } else if (key.getKind() == NodeKind.SIMPLE_VARIABLE_REF) {
+                    fieldNames.add(((BLangSimpleVarRef) key).variableName.value);
+                }
+            }
+        }
+        return fieldNames;
+    }
+
+    private void updateFieldsOfRecordLiteral(List<RecordLiteralNode.RecordField> fields, List<String> fieldNames,
```

Comment:
I would rename this method to reflect that we are adding the defaults for fields the user hasn't specified.

Reply:
Rename as `generateFieldsForUserUnspecified`